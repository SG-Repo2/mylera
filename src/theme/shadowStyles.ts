import { Platform } from 'react-native';

export const cardShadow = {
  ...Platform.select({
    ios: {
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 2 },
      shadowOpacity: 0.25,
      shadowRadius: 3.84,
    },
    android: {
      elevation: 5,
    },
  }),
}; 