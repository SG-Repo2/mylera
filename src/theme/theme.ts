import { MD3LightTheme, configureFonts } from 'react-native-paper';
import { Platform } from 'react-native';
import type { MD3TypescaleKey } from 'react-native-paper/lib/typescript/types';

// First, let's define our brand colors with semantic meaning
const brandColors = {
  primary: '#183E9F',    // Main brand color for primary actions
  secondary: '#F7A072',  // Secondary actions and highlights
  accent: '#A2D5F2',     // Subtle accents and backgrounds
  success: '#C3E8AC',    // Success states and positive feedback
  neutral: '#F5E8C7',    // Neutral backgrounds and non-interactive elements
};

// Create a complete color scheme that follows Material Design 3 principles
const colors = {
  primary: brandColors.primary,
  onPrimary: '#FFFFFF',
  primaryContainer: `${brandColors.primary}1A`, // 10% opacity version for containers
  onPrimaryContainer: brandColors.primary,
  secondary: brandColors.secondary,
  onSecondary: '#FFFFFF',
  secondaryContainer: `${brandColors.secondary}1A`,
  onSecondaryContainer: brandColors.secondary,
  tertiary: brandColors.accent,
  onTertiary: '#000000',
  tertiaryContainer: `${brandColors.accent}1A`,
  onTertiaryContainer: brandColors.accent,
  success: brandColors.success,
  onSuccess: '#000000',
  successContainer: `${brandColors.success}1A`,
  onSuccessContainer: brandColors.success,
  error: '#FF5252',
  errorContainer: '#FFCDD2',
  onError: '#FFFFFF',
  onErrorContainer: '#FF5252',
  background: brandColors.neutral,  // Back to neutral
  onBackground: '#000000',
  surface: '#FFFFFF',
  onSurface: '#000000',
  surfaceVariant: '#FFFFFF',
  onSurfaceVariant: '#000000',
  outline: '#79747E',
  elevation: {
    level0: 'transparent',
    level1: '#F5F5F5',
    level2: '#EEEEEE',
    level3: '#E0E0E0',
    level4: '#BDBDBD',
    level5: '#9E9E9E',
  },
};

// Define a complete type scale using Proxima Nova
const baseFont = {
  fontFamily: Platform.select({
    web: 'Proxima Nova',
    ios: 'Proxima Nova',
    android: 'ProximaNova',
  }) ?? 'Proxima Nova',
};

// Create a comprehensive typescale following Material Design 3 guidelines
const fontConfig = {
  displayLarge: {
    ...baseFont,
    fontSize: 57,
    lineHeight: 64,
    letterSpacing: -0.25,
    fontWeight: '400',
  },
  displayMedium: {
    ...baseFont,
    fontSize: 45,
    lineHeight: 52,
    letterSpacing: 0,
    fontWeight: '400',
  },
  displaySmall: {
    ...baseFont,
    fontSize: 36,
    lineHeight: 44,
    letterSpacing: 0,
    fontWeight: '400',
  },
  headlineLarge: {
    ...baseFont,
    fontSize: 32,
    lineHeight: 40,
    letterSpacing: 0,
    fontWeight: '700',
  },
  headlineMedium: {
    ...baseFont,
    fontSize: 28,
    lineHeight: 36,
    letterSpacing: 0,
    fontWeight: '700',
  },
  headlineSmall: {
    ...baseFont,
    fontSize: 24,
    lineHeight: 32,
    letterSpacing: 0,
    fontWeight: '700',
  },
  titleLarge: {
    ...baseFont,
    fontSize: 22,
    lineHeight: 28,
    letterSpacing: 0,
    fontWeight: '600',
  },
  titleMedium: {
    ...baseFont,
    fontSize: 16,
    lineHeight: 24,
    letterSpacing: 0.15,
    fontWeight: '600',
  },
  titleSmall: {
    ...baseFont,
    fontSize: 14,
    lineHeight: 20,
    letterSpacing: 0.1,
    fontWeight: '600',
  },
  bodyLarge: {
    ...baseFont,
    fontSize: 16,
    lineHeight: 24,
    letterSpacing: 0.15,
    fontWeight: '400',
  },
  bodyMedium: {
    ...baseFont,
    fontSize: 14,
    lineHeight: 20,
    letterSpacing: 0.25,
    fontWeight: '400',
  },
  bodySmall: {
    ...baseFont,
    fontSize: 12,
    lineHeight: 16,
    letterSpacing: 0.4,
    fontWeight: '400',
  },
  labelLarge: {
    ...baseFont,
    fontSize: 14,
    lineHeight: 20,
    letterSpacing: 0.1,
    fontWeight: '500',
  },
  labelMedium: {
    ...baseFont,
    fontSize: 12,
    lineHeight: 16,
    letterSpacing: 0.5,
    fontWeight: '500',
  },
  labelSmall: {
    ...baseFont,
    fontSize: 11,
    lineHeight: 16,
    letterSpacing: 0.5,
    fontWeight: '500',
  },
};

// Create our custom theme by extending MD3LightTheme
export const theme = {
  ...MD3LightTheme,
  colors: {
    ...MD3LightTheme.colors,
    ...colors,
  },
  fonts: configureFonts({
    config: {
      ...fontConfig,
      displayLarge: {
        ...fontConfig.displayLarge,
        fontWeight: '700' as const,
      },
      displayMedium: {
        ...fontConfig.displayMedium,
        fontWeight: '600' as const,
      },
      displaySmall: {
        ...fontConfig.displaySmall,
        fontWeight: '600' as const,
      },
      headlineLarge: {
        ...fontConfig.headlineLarge,
        fontWeight: '600' as const,
      },
      headlineMedium: {
        ...fontConfig.headlineMedium,
        fontWeight: '600' as const,
      },
      headlineSmall: {
        ...fontConfig.headlineSmall,
        fontWeight: '600' as const,
      },
      titleLarge: {
        ...fontConfig.titleLarge,
        fontWeight: '600' as const,
      },
      titleMedium: {
        ...fontConfig.titleMedium,
        fontWeight: '600' as const,
      },
      titleSmall: {
        ...fontConfig.titleSmall,
        fontWeight: '600' as const,
      },
      bodyLarge: {
        ...fontConfig.bodyLarge,
        fontWeight: '400' as const,
      },
      bodyMedium: {
        ...fontConfig.bodyMedium,
        fontWeight: '400' as const,
      },
      bodySmall: {
        ...fontConfig.bodySmall,
        fontWeight: '400' as const,
      },
      labelLarge: {
        ...fontConfig.labelLarge,
        fontWeight: '500' as const,
      },
      labelMedium: {
        ...fontConfig.labelMedium,
        fontWeight: '500' as const,
      },
      labelSmall: {
        ...fontConfig.labelSmall,
        fontWeight: '500' as const,
      },
    }
  }),
  roundness: 8,
  animation: {
    scale: 1.0,
  },
};

// Export theme type for TypeScript support
export type AppTheme = typeof theme;