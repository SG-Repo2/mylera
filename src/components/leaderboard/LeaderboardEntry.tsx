import React, { useEffect, useRef, useState } from 'react';
import { View, Text, Image, StyleSheet, Animated } from 'react-native';
import type { LeaderboardEntry as LeaderboardEntryType } from '../../types/leaderboard';
import { theme } from '../../theme/theme';
const ANIMATION_DURATION = 300;
const DEFAULT_AVATAR = require('../../../assets/images/favicon.png');

interface Props {
  entry: LeaderboardEntryType;
  highlight?: boolean;
}

/**
 * Displays individual leaderboard row with expandable details.
 * When expanded, shows additional metrics in a grid layout.
 */
export function LeaderboardEntry({ entry, highlight }: Props) {
  const [isExpanded, setIsExpanded] = useState(false);
  const { display_name, avatar_url, total_points, metrics_completed, rank } = entry;
  
  // Animation values
  const rankAnim = useRef(new Animated.Value(rank)).current;
  const pointsAnim = useRef(new Animated.Value(total_points)).current;
  const scaleAnim = useRef(new Animated.Value(1)).current;
  const prevRankRef = useRef(rank);
  const prevPointsRef = useRef(total_points);

  // Animate when rank or points change
  useEffect(() => {
    const animations = [];
    
    // Rank changed
    if (prevRankRef.current !== rank) {
      animations.push(
        Animated.timing(rankAnim, {
          toValue: rank,
          duration: ANIMATION_DURATION,
          useNativeDriver: true,
        })
      );
      prevRankRef.current = rank;
    }
    
    // Points changed
    if (prevPointsRef.current !== total_points) {
      animations.push(
        Animated.timing(pointsAnim, {
          toValue: total_points,
          duration: ANIMATION_DURATION,
          useNativeDriver: true,
        })
      );
      
      // Add scale pulse animation
      animations.push(
        Animated.sequence([
          Animated.timing(scaleAnim, {
            toValue: 1.05,
            duration: ANIMATION_DURATION / 2,
            useNativeDriver: true,
          }),
          Animated.timing(scaleAnim, {
            toValue: 1,
            duration: ANIMATION_DURATION / 2,
            useNativeDriver: true,
          }),
        ])
      );
      
      prevPointsRef.current = total_points;
    }
    
    if (animations.length > 0) {
      Animated.parallel(animations).start();
    }
  }, [rank, total_points, rankAnim, pointsAnim, scaleAnim]);


  const renderAvatar = () => {
    if (avatar_url) {
      return (
        <Image 
          source={{ uri: avatar_url }} 
          defaultSource={DEFAULT_AVATAR}
          style={styles.avatar}
          testID="avatar-image"
        />
      );
    }
    
    return (
      <View style={styles.avatarPlaceholder} testID="avatar-placeholder">
        <Text style={[styles.avatarLetter, highlight && styles.highlightText]}>
          {display_name?.charAt(0).toUpperCase() ?? '?'}
        </Text>
      </View>
    );
  };

  return (
    <View 
      style={[styles.container, highlight && styles.highlightBackground]}
      accessibilityRole="text"
      accessibilityLabel={`${display_name}, Rank ${rank}, ${total_points} points`}
      accessibilityHint={highlight ? "This is your position on the leaderboard" : undefined}
      testID="leaderboard-entry"
    >
      <Animated.View 
        style={[
          styles.mainContent,
          { transform: [{ scale: scaleAnim }] }
        ]}
      >
        {/* Rank */}
        <View style={styles.rankContainer}>
          <Animated.Text 
            style={[
              styles.rankText, 
              highlight && styles.highlightText,
              {
                transform: [{
                  translateY: rankAnim.interpolate({
                    inputRange: [rank - 1, rank, rank + 1],
                    outputRange: [-20, 0, 20]
                  })
                }]
              }
            ]}
            accessibilityLabel={`Rank ${rank}`}
            testID="rank-text"
          >
            {rank}
          </Animated.Text>
        </View>

        {/* Avatar */}
        <View style={styles.avatarContainer}>
          {renderAvatar()}
        </View>

        {/* User Info */}
        <View style={styles.infoContainer}>
          <Text 
            style={[styles.displayName, highlight && styles.highlightText]}
            accessibilityLabel={display_name}
            testID="display-name"
          >
            {display_name}
          </Text>
          <Animated.Text 
            style={[
              styles.pointsText, 
              highlight && styles.highlightText,
              {
                transform: [{
                  translateY: pointsAnim.interpolate({
                    inputRange: [total_points - 100, total_points, total_points + 100],
                    outputRange: [-20, 0, 20]
                  })
                }]
              }
            ]}
            accessibilityLabel={`${total_points} points`}
            testID="points-text"
          >
            {total_points} pts
          </Animated.Text>
        </View>

      </Animated.View>

    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 16,
    marginBottom: 8,
    borderRadius: theme.roundness,
    backgroundColor: theme.colors.surface,
    elevation: 1,
    shadowColor: theme.colors.onSurface,
    shadowOffset: { width: 0, height: 0.5 },
    shadowOpacity: 0.05,
    shadowRadius: 1,
  },
  mainContent: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 16,
  },
  highlightBackground: {
    backgroundColor: theme.colors.primaryContainer,
  },
  rankContainer: {
    marginRight: 12,
    width: 32,
    alignItems: 'center',
  },
  rankText: {
    ...theme.fonts.titleLarge,
    color: theme.colors.onSurface,
  },
  highlightText: {
    color: theme.colors.primary,
  },
  avatarContainer: {
    marginRight: 12,
  },
  avatar: {
    width: 48,
    height: 48,
    borderRadius: 24,
  },
  avatarPlaceholder: {
    width: 48,
    height: 48,
    borderRadius: 24,
    backgroundColor: theme.colors.surfaceVariant,
    alignItems: 'center',
    justifyContent: 'center',
  },
  avatarLetter: {
    ...theme.fonts.titleLarge,
    color: theme.colors.onSurfaceVariant,
  },
  infoContainer: {
    flex: 1,
  },
  displayName: {
    ...theme.fonts.titleMedium,
    color: theme.colors.onSurface,
  },
  pointsText: {
    ...theme.fonts.bodyMedium,
    color: theme.colors.onSurfaceVariant,
  },
  metricsText: {
    ...theme.fonts.bodySmall,
    color: theme.colors.onSurfaceVariant,
  },
  expandedContent: {
    backgroundColor: theme.colors.surfaceVariant,
    paddingHorizontal: 16,
    paddingBottom: 16,
  },
  expandedTitle: {
    ...theme.fonts.titleSmall,
    color: theme.colors.onSurfaceVariant,
    marginTop: 4,
    marginBottom: 8,
  },
  metricsGrid: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
  },
  metricCard: {
    width: '48%',
    backgroundColor: theme.colors.surface,
    borderRadius: theme.roundness,
    padding: 12,
    marginBottom: 8,
    elevation: 1,
    shadowColor: theme.colors.onSurface,
    shadowOffset: { width: 0, height: 0.5 },
    shadowOpacity: 0.05,
    shadowRadius: 1,
    alignItems: 'center',
  },
  metricIcon: {
    marginBottom: 8,
  },
  metricValue: {
    ...theme.fonts.titleLarge,
    color: theme.colors.onSurface,
  },
  metricUnit: {
    ...theme.fonts.bodyMedium,
    color: theme.colors.onSurfaceVariant,
  },
  metricLabel: {
    ...theme.fonts.bodySmall,
    color: theme.colors.onSurfaceVariant,
    marginTop: 2,
  },
});
