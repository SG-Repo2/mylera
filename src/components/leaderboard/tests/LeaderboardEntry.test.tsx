import React from 'react';
import { render } from '@testing-library/react-native';
import { LeaderboardEntry } from '../LeaderboardEntry';
import { mockLeaderboardEntries } from './mocks/leaderboardData';
import { theme } from '../../../theme/theme';

describe('LeaderboardEntry', () => {
  describe('Avatar Display', () => {
    it('renders avatar image when URL is provided', () => {
      const entryWithAvatar = mockLeaderboardEntries[0]; // First entry has avatar
      const { getByTestId } = render(
        <LeaderboardEntry entry={entryWithAvatar} />
      );

      const avatar = getByTestId('avatar-image');
      expect(avatar).toBeTruthy();
    });

    it('renders fallback avatar with initial when no URL is provided', () => {
      const entryWithoutAvatar = mockLeaderboardEntries[2]; // Third entry has no avatar
      const { getByText } = render(
        <LeaderboardEntry entry={entryWithoutAvatar} />
      );

      // Should show first letter of display name
      const initial = entryWithoutAvatar.display_name[0].toUpperCase();
      expect(getByText(initial)).toBeTruthy();
    });
  });

  describe('Highlighting', () => {
    it('applies highlight styles when highlight prop is true', () => {
      const entry = mockLeaderboardEntries[0];
      const { getByTestId } = render(
        <LeaderboardEntry entry={entry} highlight={true} />
      );

      const container = getByTestId('leaderboard-entry');
      expect(container.props.style).toContainEqual(
        expect.objectContaining({
          backgroundColor: theme.colors.primaryContainer,
        })
      );
      expect(container).toBeTruthy();
    });

    it('does not apply highlight styles when highlight prop is false', () => {
      const entry = mockLeaderboardEntries[0];
      const { getByTestId } = render(
        <LeaderboardEntry entry={entry} highlight={false} />
      );

      const container = getByTestId('leaderboard-entry');
      expect(container.props.style).toContainEqual(
        expect.objectContaining({
          backgroundColor: theme.colors.surface,
        })
      );
      expect(container).toBeTruthy();
    });
  });

  describe('Content Display', () => {
    it('displays rank, name and points correctly', () => {
      const entry = mockLeaderboardEntries[0];
      const { getByTestId } = render(
        <LeaderboardEntry entry={entry} />
      );

      expect(getByTestId('rank-text')).toHaveTextContent(String(entry.rank));
      expect(getByTestId('display-name')).toHaveTextContent(entry.display_name);
      expect(getByTestId('points-text')).toHaveTextContent(`${entry.total_points} pts`);
    });
  });

  describe('Accessibility', () => {
    it('includes proper accessibility labels', () => {
      const entry = mockLeaderboardEntries[0];
      const { getByTestId } = render(
        <LeaderboardEntry entry={entry} />
      );

      const accessibleElement = getByTestId('leaderboard-entry');
      expect(accessibleElement.props.accessibilityLabel).toBe(
        `${entry.display_name}, Rank ${entry.rank}, ${entry.total_points} points`
      );
      expect(accessibleElement).toBeTruthy();
    });

    it('includes position hint for highlighted entries', () => {
      const entry = mockLeaderboardEntries[0];
      const { getByTestId } = render(
        <LeaderboardEntry entry={entry} highlight={true} />
      );

      const accessibleElement = getByTestId('leaderboard-entry');
      expect(accessibleElement.props.accessibilityHint).toBe(
        "This is your position on the leaderboard"
      );
      expect(accessibleElement).toBeTruthy();
    });
  });

  // Snapshot Tests
  it('matches snapshot with avatar', () => {
    const entryWithAvatar = mockLeaderboardEntries[0];
    const { toJSON } = render(
      <LeaderboardEntry entry={entryWithAvatar} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot without avatar', () => {
    const entryWithoutAvatar = mockLeaderboardEntries[2];
    const { toJSON } = render(
      <LeaderboardEntry entry={entryWithoutAvatar} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot when highlighted', () => {
    const entry = mockLeaderboardEntries[0];
    const { toJSON } = render(
      <LeaderboardEntry entry={entry} highlight={true} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
