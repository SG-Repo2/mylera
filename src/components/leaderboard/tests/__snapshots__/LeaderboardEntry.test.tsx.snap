// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardEntry matches snapshot when highlighted 1`] = `
<View
  accessibilityHint="This is your position on the leaderboard"
  accessibilityLabel="Alice Smith, Rank 1, 200 points"
  accessibilityRole="text"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 4,
        "elevation": 1,
        "marginBottom": 8,
        "marginHorizontal": 16,
        "shadowColor": "#1C1B1F",
        "shadowOffset": {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.05,
        "shadowRadius": 1,
      },
      {
        "backgroundColor": "#E8DEF8",
      },
    ]
  }
  testID="leaderboard-entry"
>
  <Animated.View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 16,
        },
        {
          "transform": [
            {
              "scale": AnimatedValue {
                "_listeners": [],
                "_value": 1,
              },
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "marginRight": 12,
          "width": 32,
        }
      }
    >
      <Animated.Text
        accessibilityLabel="Rank 1"
        style={
          [
            {
              "color": "#1C1B1F",
              "fontSize": 22,
              "fontWeight": "400",
            },
            {
              "color": "#6200EE",
            },
            {
              "transform": [
                {
                  "translateY": {
                    "__getValue": [Function],
                    "_interpolation": {
                      "inputRange": [
                        0,
                        1,
                        2,
                      ],
                      "outputRange": [
                        -20,
                        0,
                        20,
                      ],
                    },
                  },
                },
              ],
            },
          ]
        }
        testID="rank-text"
      >
        1
      </Animated.Text>
    </View>
    <View
      style={
        {
          "marginRight": 12,
        }
      }
    >
      <Image
        defaultSource="mocked-favicon-path"
        source={
          {
            "uri": "https://example.com/alice-avatar.jpg",
          }
        }
        style={
          {
            "borderRadius": 24,
            "height": 48,
            "width": 48,
          }
        }
        testID="avatar-image"
      />
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        accessibilityLabel="Alice Smith"
        style={
          [
            {
              "color": "#1C1B1F",
              "fontSize": 16,
              "fontWeight": "500",
            },
            {
              "color": "#6200EE",
            },
          ]
        }
        testID="display-name"
      >
        Alice Smith
      </Text>
      <Animated.Text
        accessibilityLabel="200 points"
        style={
          [
            {
              "color": "#49454F",
              "fontSize": 14,
              "fontWeight": "400",
            },
            {
              "color": "#6200EE",
            },
            {
              "transform": [
                {
                  "translateY": {
                    "__getValue": [Function],
                    "_interpolation": {
                      "inputRange": [
                        100,
                        200,
                        300,
                      ],
                      "outputRange": [
                        -20,
                        0,
                        20,
                      ],
                    },
                  },
                },
              ],
            },
          ]
        }
        testID="points-text"
      >
        200
         pts
      </Animated.Text>
    </View>
  </Animated.View>
</View>
`;

exports[`LeaderboardEntry matches snapshot with avatar 1`] = `
<View
  accessibilityLabel="Alice Smith, Rank 1, 200 points"
  accessibilityRole="text"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 4,
        "elevation": 1,
        "marginBottom": 8,
        "marginHorizontal": 16,
        "shadowColor": "#1C1B1F",
        "shadowOffset": {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.05,
        "shadowRadius": 1,
      },
      undefined,
    ]
  }
  testID="leaderboard-entry"
>
  <Animated.View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 16,
        },
        {
          "transform": [
            {
              "scale": AnimatedValue {
                "_listeners": [],
                "_value": 1,
              },
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "marginRight": 12,
          "width": 32,
        }
      }
    >
      <Animated.Text
        accessibilityLabel="Rank 1"
        style={
          [
            {
              "color": "#1C1B1F",
              "fontSize": 22,
              "fontWeight": "400",
            },
            undefined,
            {
              "transform": [
                {
                  "translateY": {
                    "__getValue": [Function],
                    "_interpolation": {
                      "inputRange": [
                        0,
                        1,
                        2,
                      ],
                      "outputRange": [
                        -20,
                        0,
                        20,
                      ],
                    },
                  },
                },
              ],
            },
          ]
        }
        testID="rank-text"
      >
        1
      </Animated.Text>
    </View>
    <View
      style={
        {
          "marginRight": 12,
        }
      }
    >
      <Image
        defaultSource="mocked-favicon-path"
        source={
          {
            "uri": "https://example.com/alice-avatar.jpg",
          }
        }
        style={
          {
            "borderRadius": 24,
            "height": 48,
            "width": 48,
          }
        }
        testID="avatar-image"
      />
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        accessibilityLabel="Alice Smith"
        style={
          [
            {
              "color": "#1C1B1F",
              "fontSize": 16,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
        testID="display-name"
      >
        Alice Smith
      </Text>
      <Animated.Text
        accessibilityLabel="200 points"
        style={
          [
            {
              "color": "#49454F",
              "fontSize": 14,
              "fontWeight": "400",
            },
            undefined,
            {
              "transform": [
                {
                  "translateY": {
                    "__getValue": [Function],
                    "_interpolation": {
                      "inputRange": [
                        100,
                        200,
                        300,
                      ],
                      "outputRange": [
                        -20,
                        0,
                        20,
                      ],
                    },
                  },
                },
              ],
            },
          ]
        }
        testID="points-text"
      >
        200
         pts
      </Animated.Text>
    </View>
  </Animated.View>
</View>
`;

exports[`LeaderboardEntry matches snapshot without avatar 1`] = `
<View
  accessibilityLabel="Bob Johnson, Rank 3, 100 points"
  accessibilityRole="text"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 4,
        "elevation": 1,
        "marginBottom": 8,
        "marginHorizontal": 16,
        "shadowColor": "#1C1B1F",
        "shadowOffset": {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.05,
        "shadowRadius": 1,
      },
      undefined,
    ]
  }
  testID="leaderboard-entry"
>
  <Animated.View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 16,
        },
        {
          "transform": [
            {
              "scale": AnimatedValue {
                "_listeners": [],
                "_value": 1,
              },
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "marginRight": 12,
          "width": 32,
        }
      }
    >
      <Animated.Text
        accessibilityLabel="Rank 3"
        style={
          [
            {
              "color": "#1C1B1F",
              "fontSize": 22,
              "fontWeight": "400",
            },
            undefined,
            {
              "transform": [
                {
                  "translateY": {
                    "__getValue": [Function],
                    "_interpolation": {
                      "inputRange": [
                        2,
                        3,
                        4,
                      ],
                      "outputRange": [
                        -20,
                        0,
                        20,
                      ],
                    },
                  },
                },
              ],
            },
          ]
        }
        testID="rank-text"
      >
        3
      </Animated.Text>
    </View>
    <View
      style={
        {
          "marginRight": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E7E0EC",
            "borderRadius": 24,
            "height": 48,
            "justifyContent": "center",
            "width": 48,
          }
        }
        testID="avatar-placeholder"
      >
        <Text
          style={
            [
              {
                "color": "#49454F",
                "fontSize": 22,
                "fontWeight": "400",
              },
              undefined,
            ]
          }
        >
          B
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        accessibilityLabel="Bob Johnson"
        style={
          [
            {
              "color": "#1C1B1F",
              "fontSize": 16,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
        testID="display-name"
      >
        Bob Johnson
      </Text>
      <Animated.Text
        accessibilityLabel="100 points"
        style={
          [
            {
              "color": "#49454F",
              "fontSize": 14,
              "fontWeight": "400",
            },
            undefined,
            {
              "transform": [
                {
                  "translateY": {
                    "__getValue": [Function],
                    "_interpolation": {
                      "inputRange": [
                        0,
                        100,
                        200,
                      ],
                      "outputRange": [
                        -20,
                        0,
                        20,
                      ],
                    },
                  },
                },
              ],
            },
          ]
        }
        testID="points-text"
      >
        100
         pts
      </Animated.Text>
    </View>
  </Animated.View>
</View>
`;
